woocommerce/woocommerce-gutenberg-products-block
# Move phone to default fields section instead of being handled inline.

This PR moves the shipping and billing phones to be in the default address sections instead of being handled inline, and removes a class of issues related to that, as well as reduce complexity of codebase.
Something like https://github.com/woocommerce/woocommerce-blocks/issues/10602 since we always handled this on our own in the code.

This also fixes an issue in which phone field was always semantically outside the other fields div, and would sometimes render as a full line despite there being space for it to be half line above.

The one change is that `set-billing-phone` and `set-shipping-phone` events are no longer fired, `set-shipping-addresss` and `set-billing-address` will fire on phone is changed, this aligns it with the rest of fields in that form.

Also right now, using locale filters, someone can reposition a phone number if they want.

This is a soft prerequisite to the custom fields project.

## Testing Instructions

1. In the editor, disable the phone number. Make sure it's no longer visible in Checkout and you can place an order without it.
2. Make the phone optional, make sure it's visible but with an optional label, and that you can place an order without filling it.
3. Make the phone required. Make sure the field is visible without the optional label. Make sure you can't place an order without filling that field, and that errors are visible.
4. In the inspector, make sure the field type is tel and not text.
5. When you fill the phone field, and uncheck the "use shipping as billing" checkbox, you should see the same value in the billing one.



## Checklist

Required:

Conditional:

## Changelog


> Semantic enhacement to the position of a phone field in Checkout.
